### DISCUSSION TITLE
Looping with await UniTask.WhenAny()

### DISCUSSION THREAD

**Discussion by SamAins** [Cysharp/UniTask#398] [Q&A]
Hello I want to use WhenAny to run though a list of tasks and set them all off. like several web calls,
and when one is done I want to take it out of the list and then do some process on it. 
then wait for the next one to complete. till there are none left.

```
List<UniTask<List<int>>> tasks = new List<UniTask<List<int>>>();
List<int> results = new List<int>();
var b = new List<int>() {0,1,2,3,4,5}

foreach (var a in b)
{
    tasks.Add(SomeUniTaskmethod(a));
}

while (tasks.Count > 0)
{
    var (index, result) = await UniTask.WhenAny(tasks);
    tasks.RemoveAt(index);

    results.Concat(result);
    Debug.Log(results.Count);
} 
```

I attempt the above but get an error.

`InvalidOperationException: Already continuation registered, can not await twice or get Status after await.`

Any help with this would be great.

**Comment by Amfitorin** [comment]
Hello. For solve this problem, you need to replace foreach cycle into while space, and re run tasks every while step

**Reply by Amfitorin** [reply]
Read documentation and find interested moment. Need use UniTaskLazy for multiple awaited tasks and use in WhenAny(Lazy.Task)
for example, your code need write in this variant

List<AsyncLazy<List<int>>> tasks = new List<AsyncLazy<List<int>>>();
List<int> results = new List<int>();
var b = new List<int>() {0,1,2,3,4,5}

foreach (var a in b)
{
    tasks.Add(UniTask.Lazy(SomeUniTaskmethod(a)));
}

while (tasks.Count > 0)
{
    var (index, result) = await UniTask.WhenAny(tasks.Select(x=>x.Task));
    tasks.RemoveAt(index);

    results.Concat(result);
    Debug.Log(results.Count);
} 

this code work
